### PR TITLE
Add Sieve of Eratosthenes to generate prime numbers upto a given number

### PR DESCRIPTION
**Pull request by Rushyendher**
Added C++ code to generate prime numbers for the given input value N.

### CODE REVIEW

File: NumberTheory/SieveOfEratosthenes.cpp [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    cin >> N;
+    if (N <= 1) {
+        cout << "Invalid value! (N should be >= 2). \n";
+        getInput(N);
+    }
+}
+
+int main()
+{
+	int N;
```
Comment (faheel) [inline NumberTheory/SieveOfEratosthenes.cpp]: Use a better name for `N`, such as `prime_limit`.

File: NumberTheory/SieveOfEratosthenes.cpp [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+}
+
+void printPrimes(vector<int> &primes)
+{
+	int length = primes.size();
+	for(int i = 0;i<length;i++)
+		cout << primes[i] << "\n";
+}
+
+void getInput(int &N)
```
Comment (faheel) [inline NumberTheory/SieveOfEratosthenes.cpp]: Would be better to change this to:
```C++
void getPrimeLimit(int &prime_limit)
```

File: NumberTheory/SieveOfEratosthenes.cpp [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    (Prime Number is a number that is divisible only by itself and 1 (e.g. 2, 3, 5..))
+*/
+
+#include <iostream>
+#include <vector>
+#include <cmath>
+
+using namespace std;
+
+vector<int> sieveOfEratosthenes(int A) {
```
Comment (faheel) [inline NumberTheory/SieveOfEratosthenes.cpp]: Change the signature to:
```C++
(const int &prime_limit)
```

File: NumberTheory/SieveOfEratosthenes.cpp [hunk 4]
```diff
@@ -0,0 +1,68 @@
+/*
+    Prime numbers:
+    Given a number N, find all prime numbers upto N ( N included )
+    (Prime Number is a number that is divisible only by itself and 1 (e.g. 2, 3, 5..))
+*/
```
Comment (faheel) [inline NumberTheory/SieveOfEratosthenes.cpp]: Add the time and space complexity as mentioned in issue #8.

**Review by faheel — CHANGES_REQUESTED** [review]
This is very good @Rushyendher. Thanks!

There are however a few changes to be made before I merge this.

**Comment by Rushyendher** [comment]
ok. Thank you

**Comment by Rushyendher** [comment]
Made the requested changes.

**Review by faheel — APPROVED** [review]

**Comment by faheel** [comment]
I did a few more changes and merged it in [this commit](../commit/850141d5ba9a9abfc6ad8382723b0836cea526a5). Thanks again @Rushyendher!

**Pull request closed without merging**